examples/tictactoe/programs/tictactoe/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+use anchor_lang::prelude::*;
+use std::str::FromStr;
+
+const BOARD_ITEM_FREE: u8 = 0; // Free slot
+const BOARD_ITEM_X: u8 = 1; // Player X
+const BOARD_ITEM_O: u8 = 2; // Player O
+
+/// Game State
+/// 0 - Waiting
+/// 1 - XMove
+/// 2 - OMove
+/// 3 - XWon
+/// 4 - OWon
+/// 5 - Draw
+
+#[program]
+pub mod tictactoe {
+    use super::*;
+
+    #[access_control(Initializedashboard::accounts(&ctx))]
+    pub fn initialize_dashboard(ctx: Context<Initializedashboard>) -> ProgramResult {
+        let dashboard = &mut ctx.accounts.dashboard;
+        dashboard.game_count = 0;
+        dashboard.address = *dashboard.to_account_info().key;
+        Ok(())
+    }
+
+    #[access_control(Initialize::accounts(&ctx))]
+    pub fn initialize(ctx: Context<Initialize>) -> ProgramResult {
+        let dashboard = &mut ctx.accounts.dashboard;
+        let game = &mut ctx.accounts.game;
+        dashboard.game_count = dashboard.game_count + 1;
+        dashboard.latest_game = *game.to_account_info().key;
+        game.player_x = *ctx.accounts.player_x.key;
+        Ok(())
+    }
+
+    #[access_control(Playerjoin::accounts(&ctx))]
+    pub fn player_join(ctx: Context<Playerjoin>) -> ProgramResult {
+        let game = &mut ctx.accounts.game;
+        game.player_o = *ctx.accounts.player_o.key;
+        game.game_state = 1;
+        Ok(())
+    }
+
+    #[access_control(Playermove::accounts(&ctx,x_or_o,player_move))]
+    pub fn player_move(ctx: Context<Playermove>, x_or_o: u8, player_move: u8) -> ProgramResult {
+        let game = &mut ctx.accounts.game;
+        game.board[player_move as usize] = x_or_o;
+        game.status(x_or_o);
+        Ok(())
+    }
+
+    pub fn status(ctx: Context<Status>) -> ProgramResult {
+        msg!("{}", ctx.accounts.game.game_state);
```

Comment:
Use `emit!(event)`, where an event is defined by 

````
#[event]
pub struct Status {
  game_state,
  player_o,
  player_x,
  board,
}
```

Then we can convert via the `From` trait, i.e. `

impl From<Status> for GameState {
  fn from(status: Status) -> Self {
    ...
  }
}

Using events will allow one to easily get the type on the client side.